### PR TITLE
Added question word filter

### PR DESCRIPTION
**Pull request by tanay2001**

### CODE REVIEW

**Comment by tongshuangwu** [comment]
Hi! Thanks for your neat PR. 
1. The check seems to have failed. Could you please fix this?
```
test = {'class': 'ContractionExpansions', 'inputs': {'sentence': "Alex can't afford to buy that car"}, 'outputs': {'sentence': 'Alex cannot afford to buy that car'}}
    def execute_sentence_operation_test_case(transformation, test):
        filter_args = test["inputs"]
        outputs = test["outputs"]
        perturbs = transformation.generate(**filter_args)
        for pred_output, output in zip(perturbs, outputs):
>           assert pred_output == output["sentence"]
E           TypeError: string indices must be integers
```
2. I wonder if you have considered expanding the filter to also cover broader question types (those defined in e.g. [UIUC question classification](https://cogcomp.seas.upenn.edu/Data/QA/QC/), or in the [SQuAD paper](https://arxiv.org/abs/1606.05250))? It might give us more interesting insights beyond question words!

**Comment by tanay2001** [comment]
> Hi! Thanks for your neat PR.
> 
> 1. The check seems to have failed. Could you please fix this?
> 
> ```
> test = {'class': 'ContractionExpansions', 'inputs': {'sentence': "Alex can't afford to buy that car"}, 'outputs': {'sentence': 'Alex cannot afford to buy that car'}}
>     def execute_sentence_operation_test_case(transformation, test):
>         filter_args = test["inputs"]
>         outputs = test["outputs"]
>         perturbs = transformation.generate(**filter_args)
>         for pred_output, output in zip(perturbs, outputs):
> >           assert pred_output == output["sentence"]
> E           TypeError: string indices must be integers
> ```
> 
> 2. I wonder if you have considered expanding the filter to also cover broader question types (those defined in e.g. [UIUC question classification](https://cogcomp.seas.upenn.edu/Data/QA/QC/), or in the [SQuAD paper](https://arxiv.org/abs/1606.05250))? It might give us more interesting insights beyond question words!

Thanks for the review  @tongshuangwu 
1) Will fix that issue in the next commit
2) Indeed, will extend it to cover the broader question types as stated in the SQuAD paper

File: filters/question_word/filter.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+class NumericQuestion(QuestionAnswerOperation):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = spacy.load("en_core_web_sm")
+    
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        for ans in answers:
```
Comment (kaustubhdhole) [inline filters/question_word/filter.py:97]: Please add a function docstring: https://www.python.org/dev/peps/pep-0257/

File: filters/question_word/filter.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        return super().filter(context,question,answers)
+
+
+class NumericQuestion(QuestionAnswerOperation):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = spacy.load("en_core_web_sm")
```
Comment (kaustubhdhole) [inline filters/question_word/filter.py:94]: Also use the following line to load spacy else all your filters would accumulate memory: 
`self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")`

Corresponding import:
`from initialize import spacy_nlp`

**Comment by kaustubhdhole** [comment]
I think the PR satisfies the review criteria in the current state. I also think this is a good opportunity where there could be more question types incorporated (as @tongshuangwu mentioned the UIUC taxonomy). So, I highly recommend you to do that in a subsequent PR if you are interested. 

Besides, for this PR, please add the spacy fix and the docstrings and I am happy to approve. :)

Great work @tanay2001!

File: filters/question_word/filter.py (hunk 2 above)
Comment (tanay2001) [reply filters/question_word/filter.py]: Thanks for the review @kaustubhdhole 
I have made the suggested changes and regarding the UIUC taxonomy, indeed will give that a try too

**Comment by tanay2001** [comment]
Hi @kaustubhdhole 
I am not exactly sure why the test is failing when it's passing locally, even the error message doesn't seem to originate from my code as much as I could understand , would appreciate your help !

Thanks

**Review by kaustubhdhole — APPROVED** [review]

**Pull request merged**